### ISSUE TITLE
System.Diagnostics.StackTrace is referencing wrong version of System.Runtime.Extensions

### ISSUE THREAD

**Issue by weshaggard** [dotnet/corefx#9649]
With change https://github.com/dotnet/corefx/commit/85dda1441381fc311c9ad88dd288240877ca27b4 we started referencing the live version of System.Runtime.Extensions via ProjectReference which caused the binary to end up with 4.1.0.0 assembly reference to System.Runtime.Extensions. The issue with that is UWP only supports the 4.0.10.0 version of System.Runtime.Extensions at this point. We need to change that ProjectReference so that it references a older version (in particular https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Extensions/src/redist/System.Runtime.Extensions.depproj) so that this library will work on UWP. 

cc @ericstj

**Comment by weshaggard** [comment]
For anyone that hits this and doesn't need System.Diagnostics.StackTrace directly a workaround is to add the following to your dependencies in your project.json:
```
    "System.Diagnostics.StackTrace": {
      "version":"4.0.1",
      "exclude":"all"
    },
```
The error will usually manifest itself when running in debug CoreCLR scenarios only with the following error:
```
System.IO.FileLoadException:Could not load file or assembly 'System.Runtime.Extensions, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
```

**Comment by mikem8361** [comment]
@ellismg @ericstj How do I specify version 4.0.10.0 of System.Runtime.Extensions in the project file? Or do we add back to the project.json?

**Comment by weshaggard** [comment]
You should be able to add a project reference to https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Extensions/src/redist/System.Runtime.Extensions.depproj.

**Comment by ericstj** [comment]
> How do I specify version 4.0.10.0 of System.Runtime.Extensions in the project file?

The problem here is that you need the 4.0.10.0 *facade*.  I imagine we'll want to wrap the older version of S.R.E in a depproj and reference that.

**Comment by weshaggard** [comment]
@ericstj correct me if I'm wrong but isn't that what https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Extensions/src/redist/System.Runtime.Extensions.depproj is?

**Comment by ericstj** [comment]
I'd have to look at the impl to be sure.  IIRC the netcore50 facade was different than the default coreclr one.  NETCore50 had more private impls that called into winrt whereas the default coreclr one was mostly a facade.  I could be confusing aot though.

**Comment by weshaggard** [comment]
I'm basing my logic on https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Extensions/src/System.Runtime.Extensions.builds#L23. We may need to set the correct TargetGroup to get the right asset for the different builds but I think that depproj should work.

**Comment by ericstj** [comment]
Yeah, my point is that depproj doesn't get the dnxcore50 asset out.  I double checked and the netcore50 facade is the same so that might be OK.  I still think we need a better long term solution for this problem.  FYI, here was the issue I filed last year about this https://github.com/dotnet/corefx/issues/2817.

**Comment by weshaggard** [comment]
@ericstj agreed we should fix #2817.

**Comment by mikem8361** [comment]
@weshaggard @ericstj is there any way one of you guys can also fix this one? Especially if it is dependent on #2817.

**Comment by ericstj** [comment]
Opening to track fix in release/1.0.0

**Comment by ericstj** [comment]
This update has been pushed to nuget.org. https://www.nuget.org/packages/Microsoft.NETCore.UniversalWindowsPlatform/5.2.1
